src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1337,7 +1372,7 @@ static bool DefaultUserCertificateValidationCallback(object sender, X509Certific
         /// <summary>
         /// Creates another connector and sends a cancel request through it for this connector.
         /// </summary>
-        internal void CancelRequest()
+        internal void CancelRequest(bool allowedToThrowExceptions)
```

Comment:
Seems like the default should be to not throw, so that only the path where we want exceptions should explicitly specify the parameter.

nit: Can also rename the parameter to `throwExceptions`.